### PR TITLE
Fix request/operation ID handling

### PR DESCRIPTION
**Pull request by victorges**
We were wrongfully using the x-request-id header
as what was actually an x-operation-id.

The difference is:
 - X-request-id should change for every *request* and
   only be used within a single request on proxy/servers
   to compile logs related to the same request
 - X-operation-id is the one that should be repassed for
   every sub-request, in order to keep a correlation of
   logs between different services serving different but
   related requests

For more info ask :rodrigao:

### CODE REVIEW

**Comment by Rodrigo-Andrade** [comment]
In GoCommerce we also have something like request-hops, to know by how many servers the request needed to be processed to be fulfilled.

File: clients/ioHeadersRecorder.go [hunk 1]
```diff
@@ -31,17 +31,20 @@ func NewIOHeadersRecorder(parent *http.Request) *IOHeadersRecorder {
 	}
 
 	enableTrace, _ := strconv.ParseBool(headers.Get(enableTraceHeader))
-	requestID := headers.Get(requestIDHeader)
-	if requestID == "" {
-		requestID = uuid.NewV4().String()
+	operationID := headers.Get(operationIDHeader)
+	if operationID == "" {
+		id, err := uuid.NewV4()
```
Comment (Rodrigo-Andrade) [inline clients/ioHeadersRecorder.go]: is this compatible with dotnet guids?

**Comment by igorframos** [comment]
> In GoCommerce we also have something like request-hops, to know by how many servers the request needed to be processed to be fulfilled.

This is actually a very interesting idea. 😮

File: clients/ioHeadersRecorder.go (hunk 1 above)
Comment (victorges) [reply clients/ioHeadersRecorder.go]: I think so, it's compliant to the UUID RFCs: https://github.com/satori/go.uuid

**Pull request merged**